### DISCUSSION TITLE
What is the minimum minio deployment that supports write HA during upgrades?

### DISCUSSION THREAD

**Discussion by dedeira** [minio/minio#20648] [Q&A] [answered]
Hi,

I noticed in another discussion (https://github.com/minio/minio/discussions/18572) that practically the minimum for distributed minio is 2 nodes. Would that ensure write HA? Or it needs 3 for quorum?
My use-case is that I want to deploy minio in a kube cluster that runs over vmware in order to save logs in it. The volumes are handled by vmware and backed up periodically and I have no control on which physical drive the volumes would end up(they may all end up on the same).
So data recovery/read HA is not really my concern nor do I think I need parity data(erasure codes).
What I would like is to no lose incoming data while updating or other  maintenance on minio.
So for this, would 2 servers with one drive each suffice?

Thanks a lot!

**Comment by klauspost** [comment]
What you propose and HA is not compatible. You can *never* touch backend file state in a distributed setup, so your backup will likely not work.

For HA, do a proper setup with minimum 4 servers, direct attached storage. Not something where data ends up at random remote locations.

**Reply by dedeira** [reply]
The minio cluster will be in kubernetes, which runs over a vmware cluster. So the storage available to minio pods will be persistent volumes that I create in kube, that in turn will be written on the vm storage where the kube cluster runs. And where/how is that mapped I have no control. But I know it is periodically backed up meaning all vm state is saved so basically the cluster state.   So from the point of view of the minio pod I don't think it will notice a difference. And I understand the hardware is pretty well monitored so the risk of losing data is pretty small.
So that is why, from an app point of view, my concern is only with writing the data continuously.
So I would need a minimum that allows me to upgrade for example minio, one pod/node at a time, while continuing to write data to the cluster.
If 4 is the minimum I will use 4(1 drive per node, right?). But if I could do it with less, it would be better. The data is not mission critical but I would still like to do my best about it.
Thanks for the patience, I am pretty new to this.

**Reply by klauspost** [reply]
There are no rolling upgrades with MinIO. Pretty much everything you write is a bad idea. 

Read the documents on https://docs.min.io or reach out to sales@min.io to get proper guidance.

**Reply by dedeira** [reply]
Ok will do. Thanks!

**Reply by AlexZIX** [reply, marked as answer]
MinIO have the pretty command mc admin update which can update all nodes in your cluster simultaneously. I have 8 nodes for now in 4 different geographical locations and real downtime during upgrade to the new version is less than 1 minute. So just do it at night and have fun.
